Arrowgene.Ddon.GameServer/Handler/QuestCancelPriorityQuestHandler.cs

```diff
@@ -30,7 +30,12 @@ public override S2CQuestCancelPriorityQuestRes Handle(GameClient client, C2SQues
 
             Server.Database.DeletePriorityQuest(client.Character.CommonId, questId);
 
-            client.Party.QuestState.UpdatePriorityQuestList(Server, client.Party);
+            // The UI responds and sends this request even if you're not the leader,
+            // but we probably shouldn't let non-leaders mess with the PartyQuestState.
+            if (client.Party.Leader is not null && client.Party.Leader.Client == client)
+            {
+                client.Party.QuestState.UpdatePriorityQuestList(Server, client.Party);
+            }
```

Comment:
I think this should probably be checked inside the UpdatePriorityQuestList method maybe